apps/notification/views.py

```diff
@@ -71,3 +71,34 @@ def get_count(self, request, version=None):
             'total': total,
         }
         return response.Response(result)
+
+
+class AssignmentViewSet(viewsets.ModelViewSet):
+    serializer_class = AssignmentSerializer
+    permission_classes = [permissions.IsAuthenticated]
+    filterset_fields = ('project',)
+
+    def get_queryset(self):
+        return Assignment.get_for(
+            self.request.user,
+        )
+
+    def filter_queryset(self, queryset):
+        qs = super().filter_queryset(queryset)
+        project = self.request.query_params.get('project')
+        if project is not None:
+            qs.filter(project=project)
+        return qs
+
+    @action(
+        detail=False,
+        permission_classes=[permissions.IsAuthenticated],
+        url_path='status'
+    )
+    def get_status(self, request, version=None):
+        queryset = self.filter_queryset(self.get_queryset()).filter(is_done=False)
+        temp = list(queryset.values_list("id", flat=True))  # lazy evaluation of queryset
+        queryset.update(is_done=True)
+        data = Assignment.objects.filter(id__in=temp)
+        serializer = AssignmentSerializer(data, many=True)
+        return response.Response(serializer.data)
```

Comment:
- Let's use POST method here.
- I think we don't have to return the assignments for bulk in this case. @AdityaKhatri @samshara Do we?
    - We can just return the number of records we changed.

Reply:
Yes, a list of ids will be best.

Reply:
@AdityaKhatri Do we really need the list of ids? Are we storing assignments in redux? If we are re-loading after a bulk update, the count should be fine (for showing users).
Let's try to limit data when doing heavy bulk operations.

Reply:
Okay, that sounds good.
I was thinking of only updating the local state, but we can pull the data, it's fine.

Conclusion:
Only require a count of successfully updated records.

Reply:
Count it is @AdityaKhatri